lib/error.js

```diff
@@ -0,0 +1,49 @@
+/** takes an error object and serializes it to a plan object */
+function serializedError (input) {
+  if (!(input instanceof Error)) {
+    if (typeof input === 'string') {
+      const error = new Error(`attempted to serialize a non-error, string String, "${input}"`)
+      return serializedError(error)
+    }
+    const error = new Error(`attempted to serialize a non-error, ${typeof input} ${input?.constructor?.name}`)
+    return serializedError(error)
+  }
+  // different error objects store status code differently
+  // AxiosError uses `status`, other services use `statusCode`
+  const statusCode = input.statusCode ?? input.status
+  // CAUTION: what we serialize here gets add to the size of logs
+  return {
+    errorType: input.errorType ?? input.constructor.name,
+    ...(input.message ? { message: input.message } : {}),
+    ...(input.stack ? { stack: input.stack } : {}),
+    // think of this as error code
+    ...(input.code ? { code: input.code } : {}),
+    // think of this as http status code
+    ...(statusCode ? { statusCode } : {}),
+  }
+}
+
+/** takes an error returns new error keeping some custom properties */
+function safeError (input) {
```

Comment:
🤔 We call this `safeError` because it scrubs custom keys that likely have secrets, but we're not redacting any secrets in the preserved keys.

Should we have this package export this a function that an error with keys filtered *and* secrets redacted?

Reply:
you're totally right on this one, great callout. I refactored the code here to take this into account.